### PR TITLE
Tab styles for modern theme

### PR DESCRIPTION
**Pull request by Blackbaud-PaulCrowder**

### CODE REVIEW

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-theme?_sp.spa=pr-modern-theme-tabs](https://developer.blackbaud.com/skyux-theme?_sp.spa=pr-modern-theme-tabs)

**Comment by Blackbaud-PaulCrowder** [comment]
https://github.com/blackbaud/skyux-tabs/issues/58

File: src/app/public/styles/_mixins-public.scss [hunk 1]
```diff
@@ -59,9 +59,9 @@
   }
 }
 
-@mixin sky-theme-mode-dark {
-  .sky-theme-mode-dark,
-  :host-context(.sky-theme-mode-dark) {
+@mixin sky-theme-modern-dark {
```
Comment (Blackbaud-AlexKingman) [inline src/app/public/styles/_mixins-public.scss:62]: I'm assuming nothing is consuming this mixin yet, hence the name change? I did a quick search but found nothing.

File: src/app/public/styles/themes/modern/_buttons.scss [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  .sky-btn-tab {
+    @include sky-theme-modern-btn-tab;
+  }
+
+  .sky-btn-tab-selected:not(.sky-btn-tab-disabled) {
+    @include sky-theme-modern-btn-tab-selected;
+  }
+
+  .sky-btn-tab:hover:not(.sky-btn-tab-disabled) {
+    background-color: transparent;
```
Comment (Blackbaud-AlexKingman) [inline src/app/public/styles/themes/modern/_buttons.scss:19]: With current tabs, it looks like there's a hover background color of light grey. Is that changing?

File: src/app/public/styles/themes/modern/_compat/_mixins.scss [hunk 3]
```diff
@@ -0,0 +1,38 @@
+@import '../../../variables';
+
+@mixin sky-theme-modern-btn-tab {
+  font-weight: $sky-theme-modern-font-body-default-weight;
+}
+
+@mixin sky-theme-modern-btn-tab-selected {
+  background-color: transparent;
```
Comment (Blackbaud-AlexKingman) [inline src/app/public/styles/themes/modern/_compat/_mixins.scss:8]: I might be missing something simple, but shouldn't the selected `background-color` be a blue?

File: src/app/public/styles/_mixins-public.scss (hunk 1 above)
Comment (Blackbaud-PaulCrowder) [reply src/app/public/styles/_mixins-public.scss:62]: Correct. Anything theme-related is essentially beta and subject to change. It turns out the original intent was for this mixin to be nested inside the `@sky-theme-modern`, but that turned out not to work, which is why it was renamed and its purpose changed to not be nested.

File: src/app/public/styles/themes/modern/_buttons.scss (hunk 2 above)
Comment (Blackbaud-PaulCrowder) [reply src/app/public/styles/themes/modern/_buttons.scss:19]: @Blackbaud-ToddRoberts any opinion on this? The spec doesn't show a hover state.

File: src/app/public/styles/themes/modern/_compat/_mixins.scss (hunk 3 above)
Comment (Blackbaud-PaulCrowder) [reply src/app/public/styles/themes/modern/_compat/_mixins.scss:8]: Not in the new theme. A selected tab is decorated with a blue bottom border instead.

Comment (Blackbaud-AlexKingman) [reply src/app/public/styles/themes/modern/_compat/_mixins.scss:8]: Got it.

File: src/app/public/styles/_mixins-public.scss (hunk 1 above)
Comment (Blackbaud-AlexKingman) [reply src/app/public/styles/_mixins-public.scss:62]: Makes sense. Thanks for the explanation 👍

File: src/app/public/styles/themes/modern/_compat/_mixins.scss [hunk 4]
```diff
@@ -15,24 +23,69 @@
...
+    color: #fff;
+  }
+
   .sky-btn-tab-close,
   .sky-tab-header-count {
     color: #efefef;
   }
 }
+@mixin sky-theme-modern-dark-btn-tab-dropdown-item {
+  background-color: #252a2e;
```
Comment (Blackbaud-AlexKingman) [inline src/app/public/styles/themes/modern/_compat/_mixins.scss:75]: Should this background color be stored in a variable for reuse?

Comment (Blackbaud-PaulCrowder) [reply src/app/public/styles/themes/modern/_compat/_mixins.scss:75]: Eventually, but dark mode is in super-experimental mode right now and may be removed, so I'm trying not to create too many artifacts that relate to it yet.

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-theme?_sp.spa=pr-modern-theme-tabs](https://developer.blackbaud.com/skyux-theme?_sp.spa=pr-modern-theme-tabs)

File: src/app/public/styles/themes/modern/_compat/_mixins.scss (hunk 4 above)
Comment (Blackbaud-AlexKingman) [reply src/app/public/styles/themes/modern/_compat/_mixins.scss:75]: Sounds good.

**Pull request merged**